### PR TITLE
add AD authentication to login proccess

### PR DESCRIPTION
**Pull request by ekm6**

### CODE REVIEW

File: Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Controllers/EmployeesController.cs [hunk 1]
```diff
@@ -37,9 +37,10 @@ public async Task<ActionResult<Employee>> GetEmployee(int id)
 
         // POST: api/Login
         [HttpPost("Login")]
-        public async Task<ActionResult<User>> Login([FromBody] Employee employee)
+        // TODO: make loginRequest class
```
Comment (joeldevries) [inline Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Controllers/EmployeesController.cs:40]: Is this still a TODO?

File: Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Services/AuthService.cs [hunk 2]
```diff
@@ -27,22 +30,14 @@ public AuthService()
             _employeeRepository = new EmployeeRepository(new BenchmarkContext(new DbContextOptions<BenchmarkContext>()));
```
Comment (joeldevries) [inline Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Services/AuthService.cs:30]: EmployeeRepository should passed in by DI (same with BenchmarkContext). If you need help figuring out how to do that, I'll be happy to help.

File: Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Services/AuthService.cs [hunk 3]
```diff
@@ -123,31 +118,28 @@ public async Task<OperationResult<Employee>> GetEmployeeFromAccessToken(string a
             return (OperationResult<Employee>)OperationResult.Error("Logic error");
         }
 
-        public async Task<OperationResult<User>> Login([FromBody] Employee employee)
```
Comment (joeldevries) [inline Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Services/AuthService.cs:126]: Don't need [FromBody] attribute here.

File: Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Startup.cs [hunk 4]
```diff
@@ -31,6 +31,7 @@ public void ConfigureServices(IServiceCollection services)
             //scope interfaces
             services.AddScoped<IAuthService, AuthService>();
             services.AddScoped<IEmployeeRepository, EmployeeRepository>();
+            services.AddScoped<IActiveDirectoryRepository, ActiveDirectoryRepository>();
```
Comment (joeldevries) [inline Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Startup.cs:34]: There are ways to do DI without having to manually add each class. I can help with that if needed - you might have to bring in another library - I have recently used Autofac succesfully.

File: Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Controllers/EmployeesController.cs (hunk 1 above)
Comment (tonysulfaro) [reply Benchmark.WebApi/Benchmark.WebApi/Benchmark.WebApi/Controllers/EmployeesController.cs:40]: changed on last commit

**Pull request merged**